src/Compilers/CSharp/Portable/Symbols/ReferenceManager.cs

```diff
@@ -361,7 +361,7 @@ private bool CreateAndSetSourceAssemblyFullBind(CSharpCompilation compilation)
                         out modules,
                         resolutionDiagnostics);
 
-                    var assemblyBeingBuiltData = new AssemblyDataForAssemblyBeingBuilt(new AssemblyIdentity(name: SimpleAssemblyName), referencedAssemblies, modules);
+                    var assemblyBeingBuiltData = new AssemblyDataForAssemblyBeingBuilt(new AssemblyIdentity(name: SimpleAssemblyName, noThrow:true), referencedAssemblies, modules);
```

Comment:
Shouldn't we use the other overload here that has a comment 
``` // asserting constructor used by SourceAssemblySymbol:```

instead of the one with comment
```// constructor used by metadata reader:```
?